app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -392,7 +392,7 @@ private void showNextView(boolean resuming) {
         }
 
         if (activity.currentPromptIsQuestion()) {
-            if (!activity.saveAnswersForCurrentScreen(FormEntryConstants.EVALUATE_CONSTRAINTS)) {
+            if (!activity.saveAnswersForCurrentScreen(!activity.mFormController.isFormReadOnly())) {
```

Comment:
What was the old flag set for? I would have guessed this would at the very least be changed to 

`FormEntryConstants.EVALUATE_CONSTRAINTS and !activity.mFormController.isFormReadOnly()`

Reply:
@ctsims  `FormEntryConstants.EVALUATE_CONSTRAINTS` is a constant that equates to `true`, so it doesn't really make a difference.  Do you want to include it more for semantics ?

Reply:
FYI @ctsims

Reply:
@shubham1g5 I think the best way to say it is that either the `FormEntryConstants.EVALUATE_CONSTRAINTS` does something (in which case we should leave it in the code) or that it's antiquated and we don't use it anymore, in which case we should remove the definition of the value in addition to its use here. I'm not comfortable with only doing one.

Reply:
That makes sense! I don't see any particular use of them and thought they were there just for better semantics. Definitely pro removing them. cc: @ShivamPokhriyal